shared/src/main/scala-3/org/scalamock/clazz/Mocked.scala

```diff
@@ -0,0 +1,133 @@
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package org.scalamock.clazz
+
+import org.scalamock.context.MockContext
+
+import scala.annotation.experimental
+import scala.quoted.*
+import scala.reflect.Selectable
+
+//TODO fix MockFunction#symbol
+//TODO js annotations
+class Mocked[T: Type](
+  ctx: Expr[MockContext],
+  isStub: Boolean,
+  mockName: Option[Expr[String]]
+)(using
+  val quotes: Quotes
+):
+  val utils = Mocked.Utils()
+  import utils.quotes.reflect.*
+  val parents = List(TypeTree.of[Object], TypeTree.of[Selectable], TypeTree.of[T])
+  val tpe = TypeRepr.of[T]
+  val methods: List[utils.MockedMethod] =
+    (tpe.typeSymbol.methodMembers.toSet -- TypeRepr.of[Object].typeSymbol.methodMembers).toList
+      .map(sym => sym -> tpe.memberType(sym))
+      .zipWithIndex
+      .collect { case ((sym: Symbol, t: LambdaType), idx: Int) => utils.MockedMethod(idx, sym, t, isStub)(ctx) }
+
+  @experimental def instance: Expr[T & Selectable] =
+    val symbol: Symbol = Symbol.newClass(
+      parent = Symbol.spliceOwner,
+      name = "$anon",
+      parents = parents.map(_.tpe),
+      decls = classSymbol => methods.flatMap { method =>
+        List(
+          method.definitionSymbol(classSymbol),
+          method.mockFunctionValSymbol(classSymbol)
+        )
+      },
+      selfType = None
+    )
+
+    val definition: ClassDef =
+      ClassDef(
+        cls = symbol,
+        parents = parents,
+        body = methods.flatMap(_.implementation(symbol))
+      )
+
+    Block(
+      List(definition),
+      Apply(Select(New(TypeIdent(symbol)), symbol.primaryConstructor), Nil)
+    ).asExprOf[T & Selectable]
+
+
+object Mocked:
+  class Utils(using val quotes: Quotes):
+    import quotes.reflect.*
+
+    case class MockedMethod(idx: Int, symbol: Symbol, typ: LambdaType, stub: Boolean)(ctx: Expr[MockContext]):
+      val mockOrStub = if (stub) "Stub" else "Mock"
+      val mockValName = "mock$" + symbol.name + "$" + typ.paramTypes.map(_.show(using Printer.TypeReprShortCode)).mkString("[", ",", "]")
+
+      val mockFunctionInstance: Term =
+        val types = (typ.paramTypes :+ typ.resType).map(tpe => TypeTree.ref(tpe.typeSymbol))
+        val mockFunctionClassSym = Symbol.classSymbol(s"${mockOrStub}Function${types.length - 1}")
```

Comment:
Fully qualified name was missing here, causing an error
```suggestion
        val mockFunctionClassSym = Symbol.classSymbol(s"org.scalamock.function.${mockOrStub}Function${types.length - 1}")
```

Reply:
Found this one too, fixed it